ARM-template/CreateUiDefinition.json

```diff
@@ -0,0 +1,648 @@
+                    "regex": "^(?!(?:adm|admin|audio|backup|bin|cdrom|crontab|daemon|dialout|dip|disk|fax|floppy|fuse|games|gnats|irc|kmem|landscape|libuuid|list|lp|mail|man|messagebus|mlocate|netdev|news|nobody|nogroup|operator|plugdev|proxy|root|sasl|shadow|src|ssh|sshd|staff|sudo|sync|sys|syslog|tape|tty|users|utmp|uucp|video|voice|whoopsie|www\\-data)$)\\w+$",
+                    "validationMessage": "Username must not be a reserved Ubuntu username"
+                },
+                "osPlatform": "Linux",
+                "visible": true
+            },
+            {
+                "name": "password",
+                "type": "Microsoft.Common.PasswordBox",
+                "label": {
+                    "password": "Password",
+                    "confirmPassword": "Confirm Password"
+                },
+                "toolTip": "Default password for Kubernetes cluster portal, controllor node, Kibana, etc",
+                "constraints": {
+                    "required": true,
+                    "regex": "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-\\.]).{8,}$",
+                    "validationMessage": "Password must at least 8 characters long and contain lower case, upper case, digit and special character."
+                },
+                "visible": true
+            }
+        ],
+        "steps": [
+            {
+                "name": "commonSettings",
+                "label": "Common Settings",
+                "subLabel": {
+                    "preValidation": "Required",
+                    "postValidation": "Done"
+                },
+                "bladeTitle": "Common Settings",
+                "elements": [
+                    {
+                        "name": "dnsNamePrefix",
+                        "type": "Microsoft.Common.TextBox",
+                        "label": "Dns name prefix",
+                        "defaultValue": "",
+                        "toolTip": "Dns prefix for all components, some random value will be appended to be unique",
+                        "constraints": {
+                            "required": true
+                        },
+                        "visible": true
+                    }
+                ]
+            },
+            {
+                "name": "acsSettings",
+                "label": "ACS Settings",
+                "subLabel": {
+                    "preValidation": "Required",
+                    "postValidation": "Done"
+                },
+                "bladeTitle": "ACS Settings",
+                "elements": [
+                    {
+                        "name": "agentCount",
+                        "type": "Microsoft.Common.DropDown",
+                        "label": "Agent Count",
+                        "toolTip": "The number of agents for the cluster.  This value can be from 1 to 100 (note, for Kubernetes clusters you will also get 1 or 2 public agents in addition to these seleted masters)",
```

Comment:
is this for agent node or master node? I presume is agent. so what does "in addition to these selected masters" mean?